soccer/src/soccer/strategy/coach/coach_node.cpp

```diff
@@ -122,8 +134,109 @@ void CoachNode::check_for_play_state_change() {
 
         coach_state_pub_->publish(coach_message);
 
+        publish_static_obstacles();
+
         play_state_has_changed_ = false;
     }
 }
 
+void CoachNode::field_dimensions_callback(const rj_msgs::msg::FieldDimensions::SharedPtr& msg) {
+    current_field_dimensions_ = *msg;
+    have_field_dimensions_ = true;
+    publish_static_obstacles();
+}
+
+void CoachNode::publish_static_obstacles() {
+    if (have_field_dimensions_) {
+        // Publish defense areas as rectangular area obstacles
+        /* From the old gameplay node: "The defense area, per the rules, is the box
+         in front of each goal where only that team's goalie can be in and touch the ball." */
+
+        // Create our defense area using field dimensions
+        float def_long_dist = current_field_dimensions_.penalty_long_dist / 2.0;
+        float def_short_dist = current_field_dimensions_.penalty_short_dist;
+        float line_width = current_field_dimensions_.line_width;
+        float field_length = current_field_dimensions_.length;
+
+        rj_geometry::Point o_top_left{def_long_dist + line_width, 0.0};
+
+        rj_geometry::Point o_bot_right{-def_long_dist - line_width, def_short_dist};
+
+        auto our_defense_area{std::make_shared<rj_geometry::Rect>(o_top_left, o_bot_right)};
+
+        // Create opponent defense area using field dimensions
+
+        // Sometimes there is a greater distance we need to keep:
+        // https://robocup-ssl.github.io/ssl-rules/sslrules.html#_robot_too_close_to_opponent_defense_area
+        // TODO: update this conditional. gameplay_node used different set of checks than rules
+        // imply
+        bool is_extra_dist_necessary =
+            (current_play_state_.state == PlayState::State::Stop ||
+             current_play_state_.restart == PlayState::Restart::Direct ||
+             current_play_state_.restart == PlayState::Restart::Indirect ||
+             current_play_state_.restart == PlayState::Restart::Kickoff);
```

Comment:
I think it is actually. Since we haven't updated playstate to reflect the fact that indirect kicks no longer exist, better to cover those cases than not.

Reply:
Prabhanjan was talking about kickoff, specifically, which I am removing.